src/Tasks/Microsoft.NET.Build.Tasks/ResolveFrameworkReferences.cs

```diff
@@ -39,23 +69,93 @@ protected override void ExecuteCore()
                               NormalizeVersion(kfr.TargetFramework.Version) == NormalizeVersion(new Version(TargetFrameworkVersion)))
                 .ToDictionary(kfr => kfr.Name);
 
-            List<ITaskItem> packageReferencesToAdd = new List<ITaskItem>();
+            List<ITaskItem> packagesToDownload = new List<ITaskItem>();
+            List<ITaskItem> packagesToReference = new List<ITaskItem>();
             List<ITaskItem> runtimeFrameworks = new List<ITaskItem>();
+            List<ITaskItem> targetingPacks = new List<ITaskItem>();
             List<string> unresolvedFrameworkReferences = new List<string>();
 
+            string appHostPackPattern = null;
+            string appHostPackVersion = null;
+            string appHostRuntimeIdentifiers = null;
 
             foreach (var frameworkReference in FrameworkReferences)
             {
                 KnownFrameworkReference knownFrameworkReference;
                 if (knownFrameworkReferences.TryGetValue(frameworkReference.ItemSpec, out knownFrameworkReference))
                 {
-                    TaskItem packageReference = new TaskItem(knownFrameworkReference.TargetingPackName);
-                    packageReference.SetMetadata(MetadataKeys.Version, knownFrameworkReference.TargetingPackVersion);
-                    packageReference.SetMetadata(MetadataKeys.IsImplicitlyDefined, "true");
-                    packageReference.SetMetadata("PrivateAssets", "all");
-                    packageReference.SetMetadata("Publish", "true");
+                    if (!string.IsNullOrEmpty(knownFrameworkReference.AppHostPackNamePattern))
+                    {
+                        if (appHostPackPattern == null)
+                        {
+                            appHostPackPattern = knownFrameworkReference.AppHostPackNamePattern;
+                            appHostPackVersion = knownFrameworkReference.LatestRuntimeFrameworkVersion;
+                            appHostRuntimeIdentifiers = knownFrameworkReference.AppHostRuntimeIdentifiers;
+                        }
+                        else
+                        {
+                            //  Should not happen unless there is a bug, so this message probably doesn't need to be localized
+                            Log.LogError("Multiple FrameworkReferences defined an AppHostPack, which is not supposed to happen");
```

Comment:
I wonder if throwing an `InvalidOperationException` rather than `LogError` would be better?  The former would certainly lead users to file a bug (i.e. pretty easy for users to see a SDK task crashing as "not my problem") while this might require them to figure out what this means before doing so.